doc/Gemm.md

```diff
@@ -0,0 +1,405 @@
+                                                              _c, _ldc,
+                                                              batch_size);
+#else
+```
+
+Next we have an `#if` directive for if we want to force naive `gemm` configurations. This is triggered by a cmake variable. You can see other examples like this in `arm_gpu.hpp` which does similar things for different values of the cmake variable `BLAS_MODEL_OPTIMIZATION` .
+
+```c++
+if (_M <= 128 && _N <= 128 && _K <= 128) {
+    return blas::Gemm_Launcher<
+        64, false, false, false, 64, Tile<2, 2, 8, 8>, _t_a, _t_b,
+        static_cast<int>(gemm_memory_t::no_local),
+        static_cast<int>(gemm_algorithm_t::standard),
+        static_cast<int>(gemm_vectorization_t::full), is_beta_zero, 2,
+        static_cast<int>(
+            gemm_batch_type_t::strided)>::template _select_gemm(ex, _M, _N, _K,
+                                                                _alpha, _a,
+                                                                _lda, _b, _ldb,
+                                                                _beta, _c, _ldc,
+                                                                batch_size);
+  } else {
+    return blas::Gemm_Launcher<
+        64, false, false, false, 64, Tile<8, 8, 8, 8>, _t_a, _t_b,
+        static_cast<int>(gemm_memory_t::no_local),
+        static_cast<int>(gemm_algorithm_t::standard),
+        static_cast<int>(gemm_vectorization_t::partial), is_beta_zero, 1,
+        static_cast<int>(
+            gemm_batch_type_t::strided)>::template _select_gemm(ex, _M, _N, _K,
+                                                                _alpha, _a,
+                                                                _lda, _b, _ldb,
+                                                                _beta, _c, _ldc,
+                                                                batch_size);
+  }
+#endif
+}
+```
+
+Finally we provide a targeted configuration for small sizes (if all dimensions are less than or equal to 128) and a sensible default case for all other sizes.
+
+## CMake Configurations
+
+The generation of the `Gemm`, `Gemm_Launcher` and other operation's instantiations are driven through CMake and make use of the template files and python scripts previously touched on in [the section on source code generation](#source-code-generation).
+
+The configurations to be generated, along with associated functions, are located in `cmake/CmakeFunctionHelper.cmake` and these functions are called from `src/interface/<blas_level>/CMakeLists.txt`. Configurations are provided per backend target and will be generated for each data type set during CMake configuration with the variable `BLAS_DATA_TYPES`.
+
+As an example let's look at the configurations in `CmakeFunctionHelper.cmake` for the `RCAR` target backend, inside the function `generate_blas_gemm_objects`:
+
+```cmake
+if(${TARGET} STREQUAL "RCAR") # need investigation
```

Comment:
should we keep the `#need investigation` comment?

Reply:
No I shall remove it, good catch!